xxx/regs.c

```diff
@@ -0,0 +1,46 @@
+/**
+ * get register info
+ */
+
+#include <stdint.h>
+#include "types.h"
+
+/* 32bit registers */
+
+i32 eax() {
+        /* by calling convention */
+}
+
+i32 ebx() {
+        asm("mov eax, ebx");
+}
+
+i32 ecx() {
+        asm("mov eax, ecx");
+}
+
+i32 edx() {
+        asm("mov eax, edx");
+}
+
+i32 eip() {
+        asm("__geteip: mov eax, [esp]\n"
+            "          ret\n"
+            "call __geteip");
+}
```

Comment:
GCC dislikes that because people unaware of calling convention could mess things up trying to set local C variables from inside `asm` directives... To do that, you need to set things directly to ebp+8, ebp+16, and so on (arch dependent) and this could change depending on compiler optimizations (and compiler flags). To pack local variables, the order of variables in the stack could change also.
Then, to use oldish `asm` directive best to leave it alone inside the function (as I did). Never put C and asm together inside same function...